src/@context/UseCases/index.tsx

```diff
@@ -0,0 +1,107 @@
+import Dexie, { IndexableType, Table } from 'dexie'
+import { useLiveQuery } from 'dexie-react-hooks'
+import { ReactElement, ReactNode, createContext, useContext } from 'react'
+import { DATABASE_NAME } from './_contants'
+import {
+  ROAD_DAMAGE_TABLE,
+  RoadDamageUseCaseData
+} from './models/RoadDamage.model'
+import { LoggerInstance } from '@oceanprotocol/lib'
+
+export class UseCaseDB extends Dexie {
+  roadDamages!: Table<RoadDamageUseCaseData>
+  constructor() {
+    super(DATABASE_NAME)
+    this.version(1).stores({
+      ...ROAD_DAMAGE_TABLE
+    })
+  }
+}
+
+export const database = new UseCaseDB()
+
+interface UseCasesValue {
+  createOrUpdateRoadDamage: (
+    roadDamage: RoadDamageUseCaseData
+  ) => Promise<IndexableType>
+  roadDamageList: RoadDamageUseCaseData[]
+  updateRoadDamages: (
+    roadDamages: RoadDamageUseCaseData[]
+  ) => Promise<IndexableType>
+  deleteRoadDamage: (id: number) => Promise<void>
+  clearRoadDamages: () => Promise<void>
+}
+
+const UseCasesContext = createContext(null)
+
+function UseCasesProvider({ children }: { children: ReactNode }): ReactElement {
+  const roadDamageList = useLiveQuery(() => database.roadDamages.toArray())
+
+  const createOrUpdateRoadDamage = async (
+    roadDamage: RoadDamageUseCaseData
+  ) => {
+    const exists = roadDamageList.find(
+      (row) => roadDamage.job.jobId === row.job.jobId
```

Comment:
question: Is it possible that job is undefined?

Reply:
Added a check before inserting